File: php7/src/ChainOfResponsibility/Agent/LineManager.php
<?php

declare(strict_types = 1);


namespace PetrePatrasc\ChainOfResponsibility\Agent;


use PetrePatrasc\ChainOfResponsibility\LoanApplicationRequest;
use PetrePatrasc\ChainOfResponsibility\LoanApplicationResponse;

class LineManager extends AbstractLoanAgent
{
    public function __toString(): string
    {
        return 'Line Manager';
    }

    /**
     * @inheritDoc
     */
    public function handleLoanAmount(LoanApplicationRequest $loanRequest): LoanApplicationResponse
    {
        if ($loanRequest->getAmount() < 20000) {
            return $this->approveLoanRequest();
        }

        return $this->rejectLoanRequest();
    }

}

File: java/src/ro/petrepatrasc/chainofresponsability/agent/LineManager.java
package ro.petrepatrasc.chainofresponsability.agent;

import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationRequest;
import ro.petrepatrasc.chainofresponsability.loan.LoanApplicationResponse;

/**
 * @author Petre Pătrașc <petre@dreamlabs.ro>
 */
public class LineManager extends AbstractLoanAgent {

    public static final double AGENT_THRESHOLD = 20000;

    @Override
    public LoanApplicationResponse handleLoanRequest(LoanApplicationRequest loanRequest) {
        if (loanRequest.getAmount() < this.AGENT_THRESHOLD) {
            return this.approveLoanRequest();
        }

        return this.rejectLoanRequest();
    }
}
